projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3752,19 +3751,13 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             this.excelStyleFilteringComponent.overlayComponentId = id;
             return { id, ref: undefined };
         }
-        const ref = this.createComponentInstance(IgxGridExcelStyleFilteringComponent);
+        const ref = this.viewRef.createComponent(IgxGridExcelStyleFilteringComponent, { injector: this.injector } );
```

Comment:
Okay, so I was thinking in the lines of that this use case is already covered by the overlay service too. As in something like `this.overlayService.attach(IgxGridExcelStyleFilteringComponent, this.viewRef);` so the overlay will manage the component ref - both create (available on opening's args) and will destroy it as well, so you won't have to.
And if there's some more handling done around this - even better.